test/integration/verify/controls/generic_external_vm.rb

```diff
@@ -0,0 +1,75 @@
+
+title 'External Virtual Machine Properties'
+
+control 'azure-generic-vm-linux-external-2.0' do
+
+  impact 1.0
+  title 'Ensure External VM was built with the correct Image and has the correct properties'
+
+  # Ensure that the virtual machine has been created with the correct attributes
+  describe azure_generic_resource(group_name: 'Inspec-Azure',
+                          name: 'Linux-External-VM') do
+
+    its('location') { should cmp 'westeurope' }
+
+    # check the storage profile for the machine
+    # This can be accomplished in two ways, using dotted notation or by going into
+    # the class(es) with nested describe block
+    # METHOD 1
+    its('properties.storageProfile.imageReference.publisher') { should cmp 'Canonical' }
+    its('properties.storageProfile.imageReference.offer') { should cmp 'UbuntuServer' }
+    its('properties.storageProfile.imageReference.sku') { should cmp '16.04.0-LTS' }
+
+    # METHOD 2
+    # The `described_class` object is what is being tested and it can be used with 
```

Comment:
@adamleff is this an approach we want to document and support?

Reply:
This is most definitely something that we need to document and support.
The generic _InSpec_ resource was created so that *any* _Azure_ resource can be tested. That is anything in Azure that is managed by Resource Manager
we will never get to a point where InSpec is in step with what MS publish to Azure. So by having the generic resource it allows people to test their environments for things that we do not support explicitly yet.
This is a very common pattern for tools, for example TerraForm.
It should also mean that we get more people using the Inspec Azure resource pack because they can use it immediately without having to wait for the explicit resource for what they want to test.

Reply:
Documenting the `azure_generic_resource` ability is totally great and I'm 100% in support of it. I am not in support of having to teach users the `described_class` approach.

Reply:
Thank you, I should have been more specific: Yes, I'm very excited to see the `azure_generic_resource` Inspec resource, and I think it will be very useful, and decouples early adopters from our development cadence.   Thanks again for the idea and the implementation.

Here, though, I was specifically asking about the 'described_class' approach.

Reply:
@clintoncwolfe Fair enough, I misunderstood what you were saying here.